docs/resource_sets/resource_sets_sources/basic_example_full.cf

```diff
@@ -0,0 +1,20 @@
+
+entity Router extends std::Resource:
+    string id
+end
+
+implement Router using std::none
+
+# Router a and b are assigned to the resource set "subset_a"
+a = Router(id="router_a")
+b = Router(id="router_b")
+c = Router(id="router_c")
+std::ResourceSet(name="subset_a", resources=[a,b,c])
+
+# Router d and e are assigned to the resource set "subset_b"
+d = Router(id="router_d")
+e = Router(id="router_e")
+std::ResourceSet(name="subset_b", resources=[d,e])
+
+# Router f doesn't belong to any resource set and is shared"
+f = Router(id="router_f")
```

Comment:
This example describes that `router_f` is a shared resource. The question that can be raised here is: Why was `router_f` not added to a resource set of its own? The main reason why a resource is not added to a resource set, is because that resource is a dependency of resources in several other resource sets. I think it would be good to adjust the example in a way that reflects this. A proposal is to use the `std::AgentConfig` resource for that. This resource makes sure that an agent is started. So the idea here is that all routers in all resource sets are managed by the same agent. 

I must admit, this makes thee initial example more complex. Maybe we should exclude the shared resource completely from the example. What do you think?

```suggestion

entity Router extends std::Resource:
    string id
end

implement Router using router

implementation router for Router:
    # Resource that doesn't belong to any resource set and is shared
    std::AgentConfig(autostart=true, agentname="router_agent")
end

# Router a, b and c are assigned to the resource set "subset_a"
a = Router(id="router_a")
b = Router(id="router_b")
c = Router(id="router_c")
std::ResourceSet(name="subset_a", resources=[a,b,c])

# Router d and e are assigned to the resource set "subset_b"
d = Router(id="router_d")
e = Router(id="router_e")
std::ResourceSet(name="subset_b", resources=[d,e])
```